Auth0/CredentialsManager.swift

```diff
@@ -186,14 +186,28 @@ public struct CredentialsManager {
                                                  refreshToken: credentials.refreshToken ?? refreshToken,
                                                  expiresIn: credentials.expiresIn,
                                                  scope: credentials.scope)
-                _ = self.store(credentials: newCredentials)
-                callback(nil, newCredentials)
+                if self.willExpire(newCredentials, within: minTTL) {
+                    // TODO: On the next major add a new case to CredentialsManagerError
```

Comment:
Why do we need a new major to add another error enumeration? It would be a BC if you removed an error as this would cause existing code compilation to fail but adding one doesn't?

Reply:
Because existing switch statements in customer's code will no longer be exhaustive.